openssl/src/cms.rs

```diff
@@ -219,14 +219,66 @@ impl CmsContentInfo {
             Ok(CmsContentInfo::from_ptr(cms))
         }
     }
+
+    /// Verify this CmsContentInfo's signature, given a stack of certificates
+    /// in certs, an X509 store in store. If the signature is detached, the
+    /// data can be passed in data. The data sans signature will be copied
+    /// into output_data if it is present.
+    ///
+    /// OpenSSL documentation at [`CMS_verify`]
+    ///
+    /// [`CMS_verify`]: https://www.openssl.org/docs/manmaster/man3/CMS_verify.html
+    #[allow(clippy::redundant_clone)]
+    pub fn verify(
+        &mut self,
+        certs: Option<&StackRef<X509>>,
+        store: Option<&X509StoreRef>,
+        indata: Option<&[u8]>,
+        output_data: Option<&mut Vec<u8>>,
+        flags: CMSOptions,
+    ) -> Result<(), ErrorStack> {
+        unsafe {
+            let certs_ptr = certs.map_or(ptr::null_mut(), |p| p.as_ptr());
+            if store.is_none() && !flags.contains(CMSOptions::NO_SIGNER_CERT_VERIFY) {
```

Comment:
Does OpenSSL crash if this condition isn't met?

Reply:
Yes, it does, but not always. Only one of the tests on CI failed (see https://github.com/sfackler/rust-openssl/pull/1393#issuecomment-748515746 ) probably because input checks are not sufficient in older versions of OpenSSL.